src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/HttpContextCurrentAnalyzer.cs

```diff
@@ -40,12 +44,50 @@ public override void Initialize(AnalysisContext context)
             context.EnableConcurrentExecution();
             context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
 
-            context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpressions, SyntaxKind.SimpleMemberAccessExpression);
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (context.Compilation.Language == LanguageNames.CSharp)
```

Comment:
What's the advantage here to registering compilation?

Should we retrofit other analyzers with this pattern?

Reply:
Good question - I'm not sure which is better. This will make it so only the one we care about is actually registered. Not sure how much idfference that has